Lesson18/selcan/script.js

```diff
@@ -0,0 +1,198 @@
+
+/*
+1. write a function with array and values as 2 parameters
+2. create placeholder variable to store new array
+3. loop through every values of array
+4. use if / else condition to find if current value is equal to the value param
+5. return the result
+*/
+function removeValueFromArray(array, valueToRemove){
+  let newArray = [];
+
+  for(let i = 0; i < array.length; i++) {
+    console.log(i, array[i], valueToRemove);
+    if (array[i] !== valueToRemove){
+      console.log('inside if statemnt', i, array[i], valueToRemove);
+      newArray.push(array[i]); 
+      // 'a' !== 'b' => ['a']
+      // 'b' === 'b' => ['a']
+      // 'c' !== 'b' => ['a', 'c']
+    }
+  }
+
+  return newArray;
+}
+
+console.log("exercise 6:",removeValueFromArray(testArrayEx6, 3), removeValueFromArray(['a', 'b', 'c'], 'b'));
+console.log('----------------------------');
+// 7. Write a function that takes an object and returns the keys as an array.
+// Input: {name: "John", age: 30, city: "New York"}
+// Output: ["name", "age", "city"]
+const testObjEx7 = {name: "John", age: 30, city: "New York", isStudent: false};
+console.log(Object.keys(testObjEx7));
+
+console.log("exercise 7:" ,Object.keys(testObjEx7));
+console.log('----------------------------');
+// 8. Write a function that takes an object and returns the values as an array.
+// Input: {name: "John", age: 30, city: "New York"}
+// Output: ["John", 30, "New York"]
+const testObjEx8 = {name: "John", age: 30, city: "New York"};
+
+console.log("exercise 8:", Object.values(testObjEx8));
+console.log('----------------------------');
+// 9. Write a function that takes an object and a key, and returns true if the key exists in the object, otherwise false.
+// Input: ({name: "John", age: 30, city: "New York"}, "age")
+// Output: true
+// Input: ({name: "John", age: 30, city: "New York"}, "surname")
+// Output: false
+const testObjEx9 = {name: "John", age: 30, city: "New York", number: undefined};
+function secim(arr,val){
+    return arr.hasOwnProperty(val); 
+}
+function ex9 (object, key) {
+  console.log(object[key]);
+  return object.hasOwnProperty(key);
+}
+
+ex9(testObjEx9, 'age');
+ex9(testObjEx9, 'number');
+
+
+
+console.log('----------------------------');
+// 10. Write a function that takes an object and a key-value pair, and adds the key-value pair to the object.
+// Input: ({name: "John", age: 30}, "city", "New York")
+// Output: {name: "John", age: 30, city: "New York"}
+const testObjEx10 = {name: "John", age: 30};
+testObjEx10.city = "New York";
+  //  or;
+function adding(obj,key,val){
+    return obj[key]=val;
+}
+
+
+
+console.log('----------------------------');
+// 11. Write a function that takes an object and a key, and removes the key-value pair from the object.
+// Input: ({name: "John", age: 30, city: "New York"}, "city")
+// Output: {name: "John", age: 30}
+const testObjEx11 = {name: "John", age: 30, city: "New York"};
+ function deleting(object,keys){
+  let sonuc = delete object[keys] ;
```

Comment:
The 'delete' operator returns 'true' or 'false', see more info here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete

Instead, you need to return object ('return object;')  here.
Don't forget to call the function and print the result to the console!